code/game/mecha/mecha.dm

```diff
@@ -105,6 +104,7 @@
 	mechas_list += src //global mech list
 	reset_icon()
 	return
```

Comment:
You didn't do it properly because this return will end the proc before getting to your new line.